### PR TITLE
Added Variables for mediatypes from kitsu

### PR DESCRIPTION
**Pull request by ahmubashshir**
Added:
	Manga,Manhua,Novel,Oneshot,...
	GTK:Show mediatype in search.

### CODE REVIEW

File: trackma/ui/gtk/searchwindow.py [hunk 1]
```diff
@@ -200,9 +200,10 @@ def append(self, show):
         row = [
             str(show['id']),
             str(show['title']),
-            str(show['type']),
+            str(utils.TYPE[show['type']]),
             str(show['total']),
             color]
+        print(row)
```
Comment (FichteFoll) [inline trackma/ui/gtk/searchwindow.py]: `print` here.

Comment (ahmubashshir) [reply trackma/ui/gtk/searchwindow.py]: Sorry.😓

File: trackma/lib/libkitsu.py [hunk 2]
```diff
@@ -489,7 +495,26 @@ def _parse_info(self, media):
...
+            'manhua':   utils.TYPE_MANHUA,
+            'manhwa':   utils.TYPE_MANHWA,
+            'novel':    utils.TYPE_NOVEL,
+            'oel':      utils.TYPE_OEL,
+            'oneshot':  utils.TYPE_ONESHOT,
+        }
+        try:
+            attr_type=Type[attr['subtype']]
+        except KeyError as e:
+            print('No such Type: %s'%e.args[0])
```
Comment (dyskette) [inline trackma/lib/libkitsu.py]:
```suggestion
            print('No such Type: %s' % e.args[0])
```

File: trackma/utils.py [hunk 3]
```diff
@@ -35,14 +35,45 @@
...
+TYPE_DOUJIN = 7
+TYPE_MANGA = 8
+TYPE_MANHUA = 9
+TYPE_MANHWA = 10
+TYPE_NOVEL = 11
+TYPE_OEL = 12
+TYPE_ONESHOT = 13
+TYPE_OTHER = -1
+
+TYPE=[
```
Comment (dyskette) [inline trackma/utils.py]:
```suggestion
TYPE = [
```

File: trackma/utils.py [hunk 4]
```diff
@@ -54,7 +85,7 @@
 SEASON_SPRING = 2
 SEASON_SUMMER = 3
 SEASON_FALL = 4
-
+SEASON=[ 'All','Winter','Spring','Summer','Fall' ]
```
Comment (dyskette) [inline trackma/utils.py]:
```suggestion
SEASON = ['All', 'Winter', 'Spring', 'Summer', 'Fall']
```

Also, where is this used?

File: trackma/utils.py [hunk 5]
```diff
@@ -35,14 +35,45 @@
 STATUS_FINISHED = 2
 STATUS_NOTYET = 3
 STATUS_CANCELLED = 4
-STATUS_OTHER = 100
+STATUS_OTHER = -1
+
+STATUS=[ 'Unknown' , 'Airing' , 'Finished' , 'Not yet aired' , 'Cancelled' , 'Others' ]
```
Comment (dyskette) [inline trackma/utils.py]:
```suggestion
STATUS = ['Unknown', 'Airing', 'Finished', 'Not yet aired', 'Cancelled', 'Others']
```

Where is this used?

File: trackma/utils.py [hunk 6]
```diff
@@ -35,14 +35,45 @@
...
+STATUS_MANGA=[ 'Unknown' , 'Publishing' , 'Finished' , 'Not yet published' , 'Cancelled' , 'Others' ]
```
Comment (dyskette) [inline trackma/utils.py]:
```suggestion
STATUS_MANGA = ['Unknown', 'Publishing', 'Finished', 'Not yet published', 'Cancelled', 'Others']
```
Same, where is this used?

File: trackma/lib/libkitsu.py [hunk 7]
```diff
@@ -489,7 +495,26 @@ def _parse_info(self, media):
             total = attr['chapterCount']
         elif media['type'] == 'drama':
             total = attr['episodeCount'] # TODO Unconfirmed
-
+        Type = {
```
Comment (dyskette) [inline trackma/lib/libkitsu.py]:
```suggestion
    
        Type = {
```

File: trackma/lib/libkitsu.py [hunk 8]
```diff
@@ -489,7 +495,26 @@ def _parse_info(self, media):
...
+            'special':  utils.TYPE_SP,
+            'music':    utils.TYPE_MUSIC,
+            'manga':    utils.TYPE_MANGA,
+            'doujin':   utils.TYPE_DOUJIN,
+            'manhua':   utils.TYPE_MANHUA,
+            'manhwa':   utils.TYPE_MANHWA,
+            'novel':    utils.TYPE_NOVEL,
+            'oel':      utils.TYPE_OEL,
+            'oneshot':  utils.TYPE_ONESHOT,
+        }
```
Comment (dyskette) [inline trackma/lib/libkitsu.py]:
```suggestion
        }
        
```

File: hooks/tweet.py [hunk 9]
```diff
@@ -17,8 +17,11 @@
 
 CONSUMER_KEY    = "9Hb6ZdMmvuAWxi4GCkfhToRiH"
 CONSUMER_SECRET = "86kx9Mv9wJ5UTkDEw2jRBFYstpkDK2iP7ZAo12fhf0WooMln5w"
-
-import twitter
+try:
+    import twitter
+except ModuleNotFoundError:
+    print("tweet-hook: python3-twitter is not installed.")
+    return 1
```
Comment (dyskette) [inline hooks/tweet.py]: This looks OK, but it should be in another pull request, because this was about Kitsu's mediatypes.

File: trackma/ui/gtk/window.py [hunk 10]
```diff
@@ -357,8 +357,9 @@ def _on_about(self, action, param):
         about.set_copyright("© z411, et al.")
         about.set_authors(["See AUTHORS file"])
         about.set_artists(["shuuichi"])
-        about.connect('destroy', self._on_modal_destroy)
-        about.present()
+        about.connect('destroy-event', self._on_modal_destroy)
+        about.connect('response',lambda x,y: about.destroy())
+        about.run()
```
Comment (dyskette) [inline trackma/ui/gtk/window.py]: This should be in another pull request.

Also, it should append before run, so it'll be something like this instead:

```diff
         about.set_authors(["See AUTHORS file"])
         about.set_artists(["shuuichi"])
         about.connect('destroy', self._on_modal_destroy)
-        about.present()
+        about.connect('response', lambda dialog, response: dialog.destroy())
         self._modals.append(about)
+        about.run()

     def _on_modal_destroy(self, modal_window):
         self._modals.remove(modal_window)
```

File: hooks/tweet.py (hunk 9 above)
Comment (ahmubashshir) [reply hooks/tweet.py]: How can I add a commit in a different PR?

File: trackma/utils.py (hunk 4 above)
Comment (ahmubashshir) [reply trackma/utils.py]: We can use them to show String instead of a Number.

File: trackma/utils.py (hunk 5 above)
Comment (ahmubashshir) [reply trackma/utils.py]: We can use them to show String instead of a Number.

**Comment by ahmubashshir** [comment]
Should I remove currently unused variables?

File: hooks/tweet.py (hunk 9 above)
Comment (FichteFoll) [reply hooks/tweet.py]: Is this a return at module level? (Amon mobile.)

Comment (ahmubashshir) [reply hooks/tweet.py]: Fixed in 41c6b4b.

**Comment by FichteFoll** [comment]
This PR is becoming a bit messy now because you started it from your `master` branch, meaning all further commits to your `master` branch end up as suggested changes in this PR.

For the future, you should start work on a feature or bugfix on a separate branch and only include relevant changes/commits there. When you start working on something new, e.g. #177, create a new branch and PR from it.

**Comment by ahmubashshir** [comment]
Thanks for the suggestion. @FichteFoll

**Comment by ahmubashshir** [comment]
@dyskette @FichteFoll should I close this pr and and commit in different branch then open pr?

**Comment by FichteFoll** [comment]
Imo, yes, but @z411 should be the person to ask.

**Comment by ahmubashshir** [comment]
@dyskette , @FichteFoll will you please review #437 and #436.
